server/src/test/java/org/opensearch/rest/ResponseLimitSettingsTests.java

```diff
@@ -0,0 +1,221 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.rest;
+
+import org.opensearch.Version;
+import org.opensearch.cluster.ClusterState;
+import org.opensearch.cluster.metadata.IndexMetadata;
+import org.opensearch.cluster.metadata.Metadata;
+import org.opensearch.cluster.routing.IndexRoutingTable;
+import org.opensearch.cluster.routing.IndexShardRoutingTable;
+import org.opensearch.cluster.routing.RoutingTable;
+import org.opensearch.cluster.routing.ShardRouting;
+import org.opensearch.cluster.routing.ShardRoutingState;
+import org.opensearch.cluster.routing.TestShardRouting;
+import org.opensearch.common.settings.ClusterSettings;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.core.index.Index;
+import org.opensearch.core.index.shard.ShardId;
+import org.opensearch.test.OpenSearchTestCase;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.opensearch.rest.ResponseLimitSettings.CAT_INDICES_RESPONSE_LIMIT_SETTING;
+import static org.opensearch.rest.ResponseLimitSettings.CAT_SEGMENTS_RESPONSE_LIMIT_SETTING;
+import static org.opensearch.rest.ResponseLimitSettings.CAT_SHARDS_RESPONSE_LIMIT_SETTING;
+
+public class ResponseLimitSettingsTests extends OpenSearchTestCase {
+
+    public void testIsResponseLimitBreached_forNullMetadata_expectNotBreached() {
+        final Settings settings = Settings.builder().build();
+        final ClusterSettings clusterSettings = new ClusterSettings(Settings.EMPTY, ClusterSettings.BUILT_IN_CLUSTER_SETTINGS);
+        final ResponseLimitSettings responseLimitSettings = new ResponseLimitSettings(clusterSettings, settings);
+        final boolean breached = ResponseLimitSettings.isResponseLimitBreached(
+            (Metadata) null,
+            ResponseLimitSettings.LimitEntity.INDICES,
+            0
+        );
+        assertFalse(breached);
+    }
+
+    public void testIsResponseLimitBreached_forNullRoutingTable_expectNotBreached() {
+        final Settings settings = Settings.builder().build();
+        final ClusterSettings clusterSettings = new ClusterSettings(Settings.EMPTY, ClusterSettings.BUILT_IN_CLUSTER_SETTINGS);
+        final ResponseLimitSettings responseLimitSettings = new ResponseLimitSettings(clusterSettings, settings);
+        final boolean breached = ResponseLimitSettings.isResponseLimitBreached(
+            (RoutingTable) null,
+            ResponseLimitSettings.LimitEntity.INDICES,
+            0
+        );
+        assertFalse(breached);
+    }
+
+    public void testIsResponseLimitBreached_forCatIndicesWithSettingDisabled_expectNotBreached() {
```

Comment:
- avoid underscore in method names.
- nit : long method names can  be `tescatIndicesWithSettingDisabled`